Golem friendly fire

ikonomov opened the pull request:
Makes Golems immune to friendly fire when friendly fire setting is disabled and friendly mode is enabled.  Fixes https://github.com/diasurgical/devilutionX/issues/4352

ikonomov committed:
Golem friendly fire

Makes Golems immune to friendly fire when friendly fire setting is disabled and friendly mode is enabled.

Chonkblonk commented:
I like this idea, it would make the Golem much more useful in general, rather than just whenever Sorcerer runs into dual immunes on higher difficulties (together with Stone Curse).

kphoenix137 commented:
A further improvement could be made by entirely disabling damage vs your own golem, while toggling damage vs other players' golems using friendly fire/player friendly

ikonomov commented:
Casting the spell again to unsummon the golem costs a lot of mana.  It could be a combination though, since there are two separate conditions.  So if friendly fire setting is disabled, you can't hit your golem at all even when PvP mode is on.  This way we can have it usable both ways.

kphoenix137 commented:
> Casting the spell again to unsummon the golem costs a lot of mana. It could be a combination though, since there are two separate conditions. So if friendly fire setting is disabled, you can't hit your golem at all even when PvP mode is on. This way we can have it usable both ways.

For a low level character where costing Golem is quite costly, what are the incentives to purposefully kill your own Golem? If you could find such an incentive, other options would be to cast Town Portal or walk onto a stair tile to kill the Golem, which are not very costly at all.

ikonomov committed:
Update player.cpp

as suggested in https://github.com/diasurgical/devilutionX/pull/7010#issuecomment-1984783650

D1-Constantine commented:
> > Casting the spell again to unsummon the golem costs a lot of mana. It could be a combination though, since there are two separate conditions. So if friendly fire setting is disabled, you can't hit your golem at all even when PvP mode is on. This way we can have it usable both ways.
> 
> For a low level character where costing Golem is quite costly, what are the incentives to purposefully kill your own Golem? If you could find such an incentive, other options would be to cast Town Portal or walk onto a stair tile to kill the Golem, which are not very costly at all.

KP the Golem is used mostly to glitch out the monster AI when you are taking & defending your position in the dungeon, after that you MUST kill him otherwise he'll pull the whole map

qndel commented:
@D1-Constantine can't you kill the golem by recasting the spell though?

D1-Constantine commented:
> @D1-Constantine can't you kill the golem by recasting the spell though?

Correct, but a golem at max slvl cost what 60 mana ? That is a lot !

qndel commented:
> > @D1-Constantine can't you kill the golem by recasting the spell though?
> 
> Correct, but a golem at max slvl cost what 60 mana ? That is a lot !

That's actually a good point

kphoenix137 commented:
> > > Casting the spell again to unsummon the golem costs a lot of mana. It could be a combination though, since there are two separate conditions. So if friendly fire setting is disabled, you can't hit your golem at all even when PvP mode is on. This way we can have it usable both ways.
> > 
> > 
> > For a low level character where costing Golem is quite costly, what are the incentives to purposefully kill your own Golem? If you could find such an incentive, other options would be to cast Town Portal or walk onto a stair tile to kill the Golem, which are not very costly at all.
> 
> KP the Golem is used mostly to glitch out the monster AI when you are taking & defending your position in the dungeon, after that you MUST kill him otherwise he'll pull the whole map

https://github.com/diasurgical/devilutionX/pull/7010/files

My PR allows that by allowing you to despawn your golem at no mana cost by recasting while it's alive.
